app/Http/Controllers/TaskController.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Task;
+use Illuminate\Http\Request;
+use Illuminate\Validation\Rule;
+
+class TaskController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $tasks = Task::orderBy('created_at', 'asc')->get();
+        return view('tasks')->with([
+            'tasks' => $tasks
+        ]);
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        //
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $date = $request->date;
+        $time = $request->time;
+        $request->validate([
+            'name' => ['required', 'max:30', 'min:10'],
+            'date' => ['required', Rule::unique('tasks')->where(function ($query) use ($date, $time) {
+                return $query->where('date', $date)
+                    ->where('time', $time);
+            })],
+            'time' => ['required',]
+        ]);
+        Task::storeTask($request->all());
+        return back();
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        //
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($id)
+    {
+        $tasks = Task::orderBy('id', 'asc')->get();
+        $task = Task::find($id);
+        return view('editTask')->with(['task' => $task, 'tasks' => $tasks]);
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, $id)
+    {
+        $date = $request->date;
+        $time = $request->time;
+        $request->validate([
+            'name' => ['required', 'max:30', 'min:10'],
+            'date' => ['required', Rule::unique('tasks')->ignore($id)->where(function ($query) use ($date, $time) {
+                return $query->where('date', $date)
+                    ->where('time', $time);
+            })],
+            'time' => ['required']
+        ]);
```

Comment:
This is repeated twice in the file

Reply:
One is for store function and second is for update function

Reply:
remove this from both store and update and create new function